includes/class-wsuwp-graduate-degree-programs.php

```diff
@@ -563,4 +587,77 @@ public function save_factsheet( $post_id, $post ) {
 			}
 		}
 	}
+
+	/**
+	 * Returns a useable subset of data for displaying a factsheet.
+	 *
+	 * @since 0.4.0
+	 *
+	 * @param int $post_id
+	 *
+	 * @return array
+	 */
+	public static function get_factsheet_data( $post_id ) {
+		$factsheet_data = get_registered_metadata( 'post', $post_id );
+
+		$data = array(
+			'degree_id' => 0,
+			'description' => '',
+			'accepting_applications' => 'No',
+			'students' => 0,
+			'aided' => 0,
+			'degree_url' => 'Not available',
+			'admission_requirements',
+			'student_opportunities',
+			'career_opportunities',
+			'career_placements',
+			'student_learning_outcome',
+		);
+
+		if ( isset( $factsheet_data['gsdp_degree_description'][0] ) ) {
+			$data['description'] = $factsheet_data['gsdp_degree_description'][0];
+		}
+
+		if ( isset( $factsheet_data['gsdp_degree_id'][0] ) ) {
+			$data['degree_id'] = $factsheet_data['gsdp_degree_id'][0];
+		}
+
+		if ( isset( $factsheet_data['gsdp_accepting_applications'][0] ) && 1 === absint( $factsheet_data['gsdp_accepting_applications'][0] ) ) {
+			$data['accepting_applications'] = 'Yes';
+		}
+
+		if ( isset( $factsheet_data['gsdp_grad_students_total'][0] ) ) {
+			$data['students'] = $factsheet_data['gsdp_grad_students_total'][0];
+		}
+
+		if ( isset( $factsheet_data['gsdp_grad_students_aided'][0] ) ) {
+			$data['aided'] = round( ( $factsheet_data['gsdp_grad_students_aided'][0] / $data['students'] ) * 100, 2 );
```

Comment:
This may be a non-issue, but this will throw a "division by zero" warning if the `students` field is left empty (or as the default `0` value, that is).